### PR TITLE
Update key names in adapt keyword registration message

### LINKED ISSUE #2950: Improve naming of entity registration arguments
**Description:**
When registering entities with Adapt we use `start_concept, end_concept` whilst Adapt uses `entity_value, entity_type`. Start and End don't really reflect what the arguments are (if I understand them correctly). So I'm recommending that we rename them throughout the code base. Docstrings should also be added or updated as needed.

**Additional context**
Originally raised in this PR
https://github.com/MycroftAI/mycroft-core/pull/2877#pullrequestreview-670931419

### PR DESCRIPTION
**Pull request by forslund**
## Description
This should resolve #2950. It matches the keyword entity terms in Mycroft with the ones used in Adapt. 

This changes the internally used names for entities and entity values when
sent on the messagebus and used internally in the intent service from start / end to entity_value and entity_type.

This makes the terminology easier to understand and follow across into Adapt.

The old terms are still included and usable for compatibility but should be
removed in an upcoming major release (22.02).

## How to test
Ensure VK tests works.

## Contributor license agreement signed?
CLA [ Yes ]

### CODE REVIEW

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/core/PR-2963)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/core/PR-2963/logs/skills.log), [audio.log](https://reports.mycroft.ai/core/PR-2963/logs/audio.log), [voice.log](https://reports.mycroft.ai/core/PR-2963/logs/voice.log), [bus.log](https://reports.mycroft.ai/core/PR-2963/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/core/PR-2963/logs/enclosure.log)

**Comment by forslund** [comment]
I'm doing some work on a skill implementation in rust and used the updated names here but realized it hasn't been reviewed. Should I revert back to the current names or is this something that could be included down the line?

File: mycroft/skills/intent_service.py [hunk 1]
```diff
@@ -554,3 +560,29 @@ def handle_entity_manifest(self, message):
         self.bus.emit(message.reply(
             "intent.service.padatious.entities.manifest",
             {"entities": self.padatious_service.registered_entities}))
+
+
+def is_old_style_keyword_message(message):
```
Comment (JarbasAl) [inline mycroft/skills/intent_service.py:565]: should this be private since it will be dropped soon? just be sure nothing uses it in the wild

Comment (forslund) [reply mycroft/skills/intent_service.py:565]: That's a good idea. will update

**Review by krisgesling — APPROVED** [review]
Yeah, great clean up - thanks!

**Pull request merged**